portals/publisher/src/main/webapp/source/src/app/components/GlobalPolicies/Policies/GlobalSpecificComponents/General.tsx

```diff
@@ -0,0 +1,458 @@
+            else if (spec.type.toLowerCase() === 'boolean') return (previousVal.toString() === 'true');
+            else return previousVal;
+        } else if (spec.defaultValue !== null && spec.defaultValue !== undefined) {
+            if (spec.type.toLowerCase() === 'integer') return parseInt(spec.defaultValue, 10);
+            else if (spec.type.toLowerCase() === 'boolean') return (spec.defaultValue.toString() === 'true');
+            else return spec.defaultValue;
+        } else {
+            return '';
+        }
+    }
+
+    /**
+     * Reset the input fields
+     */
+    const resetAll = () => {
+        setState(initState);
+    }
+
+    /**
+     * Function to check whether there are any errors in the form.
+     * If there are errors, we disable the save button.
+     * @returns {boolean} - Boolean value indicating whether or not the form has any errors.
+     */
+    const formHasErrors = () => {
+        let formHasAnError = false;
+        policySpec.policyAttributes.forEach((spec) => {
+            if(getError(spec) !== '') {
+                formHasAnError = true
+            }
+        })
+        return formHasAnError;
+    }
+
+    /**
+     * Function to check if the form content is in state that needs to be saved.
+     * @returns {boolean} - Whether or not the save button should be disabled.
+     */
+    const isSaveDisabled = () => {
+        if (!isEditMode) {
+            let isDisabled = false;
+            policySpec.policyAttributes.forEach((spec) => {
+                if (spec.type !== 'Boolean') {
+                    const currentState = state[spec.name];
+                    const currentVal = getValue(spec);
+                    if (spec.required && !(currentState || currentVal)) {
+                        isDisabled = true;
+                    }
+                }
+            });
+            return isDisabled;
+        } else {
+            let isDisabled = true;
+            policySpec.policyAttributes.forEach((spec) => {
+                if (spec.type !== 'Boolean') {
+                    const currentState = state[spec.name];
+                    if (currentState !== null) {
+                        isDisabled = false;
+                    }
+                } else {
+                    const currentState = state[spec.name];
+                    if (
+                        currentState !== null &&
+                        (currentState.toString() === 'true' ||
+                            currentState.toString() === 'false')
+                    ) {
+                        isDisabled = false;
+                    }
+                }
+            });
+            return isDisabled;
+        }
+    };
+
+    const hasAttributes = policySpec.policyAttributes.length !== 0;
+    const resetDisabled = Object.values(state).filter((value: any) => 
+        (value !== null && (value.toString() !== 'true' || value.toString() !== 'false')) || !!value
```

Comment:
Do we have any particular reason to add 2 negations to value as `!!value`?

Reply:
According to my knowledge, In Javascript, 2 negations used to convert a value to its corresponding boolean representation. This same line is also used API Policies so I reused it in my code.